### PR TITLE
Extracts re-usable workflows adds nightly notifications

### PR DESCRIPTION
**Pull request by kelsos**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

I am also opening the PR to check the regular CI flow that I didn't check before

### CODE REVIEW

**Comment by kelsos** [comment]
## Enabling pytest on Windows.

The reusable workflow for the backend is windows ready, however, multiple tries yesterday proved that running on windows is extremely slow.

While trying different options and arguments the tests manage to get from 16%-20% of the test suite in 1h10 minutes.

For this reason the windows nightly was removed.

## TODO

To activate windows we need to figure out:
1. If this is a CI situation or also happens in a Windows PC installation.
2. Figure out what causes the tests to be extremely slow and fix it.

> **Note**! Someone should still check the full test suite on a Windows machine since there were a number of errors that didn't appear on Linux/macOS (a lot of them related to the filesystem/temporary files)

**Review by yabirgb — COMMENTED** [review]
2 comments + 1 neatpick

File: .github/scripts/notifier.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+import argparse
+from http import HTTPStatus
+
+import requests
+
+ALL_DEVS = "1044950703795359756"
+BACKEND_DEVS = "1044951412674674758"
```
[the remark this answers is not in the document]
Comment (kelsos) [reply .github/scripts/notifier.py:9]: image

File: .github/scripts/notifier.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        group = ALL_DEVS
+        emoji = ':construction_site:'
+        job_type = 'builds'
+
+    data = {
+        "content": f'{emoji} **Github Actions:** {job_type} failed :x: \r\n\r\n '
+                   f'<@&{group}> please have a look at '
+                   f'[{run_id}](https://github.com/rotki/rotki/actions/runs/{run_id}) :cry:',
+    }
+
```
[the remark this answers is not in the document]
Comment (kelsos) [reply .github/scripts/notifier.py:47]: We can do it in the PR that changes the ids too

File: .github/workflows/rotki_dev_builds.yml [hunk 3]
```diff
@@ -303,12 +277,26 @@ jobs:
...
+      - name: Check if any task failed
         run: |
-          curl --request POST \
-            --url ${{ secrets.WEBHOOK }} \
-            --header 'Content-Type: application/json' \
-            --data '{
-          	"content": ":construction_site: **Github Actions:** ${{ github.job }} failed :x: \r\n\r\n  <@&735068019440615516> please have a look at [${{ github.run_id }}](https://github.com/rotki/rotki/actions/runs/${{ github.run_id }}) :cry:"
-          }'
+          pip install requests
+
```
[the remark this answers is not in the document]
Comment (kelsos) [reply .github/workflows/rotki_dev_builds.yml:294]: It is already cached, we can change it in the future but it won't probably haver any major impact

**Review by yabirgb — APPROVED** [review]
Requested changes will happen in a following PR so merging for now as I didn't see any major issue unknown

**Pull request merged**